egs/wsj/s5/steps/libs/nnet3/train/common.py

```diff
@@ -771,6 +771,11 @@ def __init__(self):
                                  lstm*=0,0.2,0'.  More general should precede
                                  less general patterns, as they are applied
                                  sequentially.""")
+        self.parser.add_argument("--trainer.dropout-per-frame", type=str,
```

Comment:
Is this option required? Do you expect to change whether dropout is per frame or not during the training iterations?
I think dropout-per-frame should only be at the config level. 
Also I think you can remove dropout_per_frame from the function `SetDropoutProportion`, because that is something you would have already defined from the config. If you really need to change dropout-per-frame during training, I suggest add a separate function like SetDropoutPerFrame to the DropoutComponent.